README.md

```diff
@@ -68,6 +68,105 @@ poetry run flake8
 poetry run isort -rc .
 ```
 
+## How to Test Integration With Slack
+
+After having developed some new feature, or having in hand what you believe is
+a fix for an existing bug, how do you test it out in a real system in order to
+make sure that your changes do all that you hope they do? The answer; bring up
+the application in your own environment and hook it up to Slack!
+
+In order to do this, there are 6 main things you need to accomplish:
+
+1. Setup your own Slack workspace.
+2. Grab a signing secret from Slack that pybot can utilize.
+3. Launch pybot locally, passing it your Slack signing secret.
+4. Attach your pybot instance to the public internet so that Slack can speak
+   with it.
+5. Point Slack at your running pybot instance, and properly configure it.
+6. Relish in your success!
+
+The following sections will guide you through each of these stages.
+
+### 1 - Setup Your Own Slack Workspace
+
+To start, you'll want to visit Slack's [Getting
+Started](https://slack.com/get-started) page. From this page, choose the option
+to create a new workspace. It will request that you enter in an email address,
+feel free to enter in an email address that you have access to and continue
+through the process. It will email you a 6 digit code, check your email for
+this code and then input it into Slack to continue the process. Enter in the
+name of the workspace that you're going to play around with the pybot in. When
```

Comment:
On the one hand, I like the level of detail for walking people through the process. On the other hand, I feel like Slack might change their process for setting up a workstation and then this would be outdated. I'm undecided as to which situation is better... I'll leave it up to Allen, but I just wanted to point it out.

Reply:
I have the same worry. I just don't know what else to do. The steps are complex enough that it could prevent someone from being able to follow through with setting up their environment, but they could also change in a moment's notice. The only alternative I can think of is calling out the high level idea of what needs to happen and try to point at some Slack documentation that they will hopefully keep up to date. I'm still worried about that not being good enough for our needs though.

Reply:
Yeah. So the assumption is that if we keep this level of detail, it will work for a little bit. Then when something changes, someone will try it and realize that the steps we list out are outdated. Hopefully the changes are minor enough that they figure out how to continue setting it up and then they can just make a PR to the README or tell one of us that it's outdated so we can fix it.

I would think that having too much outdated information wouldn't stop someone from contributing. And if it does, then they probably aren't advanced enough to be working on pybot. This is not really a "my first project" kind of codebase, for better or worse.

But because it's not really a "my first project" kind of codebase, I don't think we really _need_ to baby-step people through the 6 digit code and all those details. If you can't figure that out by looking at Slack's documentation on how to set up a workspace, you will find nothing but frustration and failure trying to contribute to pybot. Hence, a link to the docs is fine and if someone really wants to contribute and can't figure it out, they can ask us to pair with them on it.

What say you, @AllenAnthes?

Reply:
That sounds reasonable. We will want to explicitly call out some items of importance, particularly configuration items that must be set in order for integration to work. So perhaps generally point them at the place and documentation, and then call out something about how it should look like when done? Like; slash command x should point at URI y or some such?

Reply:
Right, anything specific to the pybot setup definitely needs to be in there. But before we remove the extra stuff, let Allen chime in. He might feel like he wants this to be more accessible to beginners and that this level of detail is the way to go

Reply:
I followed through with doing this, let me know what you think.